deployer/src/runtime_manager.rs

```diff
@@ -115,19 +131,56 @@ impl RuntimeManager {
         .await
         .context("failed to start shuttle runtime")?;
 
+        let process = Arc::new(Mutex::new(process));
+        let process_cloned = Arc::clone(&process);
         self.runtimes
             .lock()
             .unwrap()
             .insert(id, (process, runtime_client.clone()));
 
+        let stdout = process_cloned
+            .lock()
+            .await
+            .stdout
+            .take()
+            .context("child process did not have a handle to stdout")?;
+
+        // Ensure the child process is spawned in the runtime so it can
+        // make progress on its own while we await for any output.
+        tokio::spawn(async move {
+            process_cloned
+                .lock()
+                .await
+                .wait()
+                .await
+                .expect("child process encountered an error");
+        });
```

Comment:
Just wondering, is it worth waiting in a task that isn't awaited?

Reply:
From the docs:

> The provided future will start running in the background immediately when spawn is called, even if you don’t await the returned `JoinHandle`.

Reply:
Yup, 002d0c01a408f49bfd49bbe4a82f042bdb6b5847